LinqToRest/ResourceRetriever.cs

```diff
@@ -106,9 +108,42 @@ private static Uri GetResourceUri(JToken token, Uri root)
         private object GetDeserializedParameter(ParameterInfo parameter, JToken token, Uri uri)
         {
             var type = parameter.ParameterType;
+
             return type.IsQueryable()
                 ? _queryableFactory(type.GetInnerType(), uri)
-                : GetField(type, token, parameter.Name);
+                : type.IsEnum
+                    ? GetEnumFromType(type, token, parameter.Name)
+                    : GetField(type, token, parameter.Name);
+        }
+
+        private static object GetEnumFromType(Type type, JToken token, string name)
+        {
+            var getEnum = typeof(ResourceRetriever).GetMethod(nameof(GetEnum))?.MakeGenericMethod(type);
+
+            try
+            {
+                return getEnum is null
+                    ? throw new InvalidOperationException()
+                    : getEnum.Invoke(null, new object[] {token, name});
+            }
+            catch (System.Reflection.TargetInvocationException e)
+            {
+                if (e.InnerException != null)
+                {
+                    throw e.InnerException;
+                }
+                throw;
+            }
+        }
+
+        public static T GetEnum<T>(JToken token, string name) where T : struct

```

Comment:
Why does `T` need to be a struct? What happens to the reflection call above when `T` is a `class`?

Reply:
`GetField` accepts a `type`. Wouldn't it be consistent, if `GetEnum` had the same parameters instead of being a generic?

Reply:
I extracted the parse methods to own class and that cleand up the `GetEnum` function https://github.com/messerli-informatik-ag/linq-to-rest/pull/15/commits/e32cccdcc20fcceb77141a33f1d8c11c502c9661